unicore/distribute/api/repos.py

```diff
@@ -73,7 +92,85 @@ def post(self):
                     'url': self.request.route_url('repositoryresource',
                                                   name=name)
                 }))
-        return format_diffindex(changes)
+        return list(format_diffindex(changes))
+
+
+def initialize_repo_index(event):
+    repo = event.repo
+    model_mappings = map(
+        lambda (content_type, mapping): (load_model_class(repo, content_type),
+                                         mapping),
+        event.mapping.iteritems())
+    sm = StorageManager(repo)
+    im = ESManager(
+        storage_manager=sm,
+        es=get_es(event.config),
+        index_prefix=get_index_prefix(repo.working_dir))
+
+    if im.index_exists(sm.active_branch()):
+        im.destroy_index(sm.active_branch())
+    im.create_index(sm.active_branch())
```

Comment:
Indexes aren't always immediately available, sometimes ES does some thinking before it is, in tests we block on this here: https://github.com/universalcore/unicore.distribute/blob/feature/issue-19-update-index-when-content-is-modified/unicore/distribute/tests/base.py#L33